### ISSUE TITLE
luasnip configured incorrectly in others.lua

### ISSUE THREAD

**Issue by jrodal98** [NvChad/NvChad#526]
**Describe the bug**
in [others.lua](https://github.com/NvChad/NvChad/blob/6fb97be55c647ccfe2f46e3b981dbcd5f5bd8598/lua/plugins/configs/others.lua#L82), you have this:

```
require("luasnip/loaders/from_vscode").load { path = { chadrc_config.plugins.options.luasnip.snippet_path } }
```

When it should be this:

```
require("luasnip/loaders/from_vscode").load { paths = chadrc_config.plugins.options.luasnip.snippet_path }
```

Note the use of path**s** instead of path and removal of the pair of curly braces.

You can see that this is the correct usage in the lua snip repo's example file: <https://github.com/L3MON4D3/LuaSnip/blob/94bf4e7cc56d03d82c4058df7090f9d982aa264d/Examples/snippets.lua#L491>

**To Reproduce**
Assuming you haven't changed others.lua yet:

1. Download snippets directory from here: https://github.com/jrodal98/nvim/tree/c62f2b5850fcb17277b70461fc10df5d4ba3df1e/lua/custom
2. set `snippet_path = {"./lua/custom/snippets"},` in chadrc
3. open a python file
4. Type `shebang`. My `shebang` snippet I added doesn't show up

**Expected behavior**
The shebang snippet I added should show up

**Desktop (please complete the following information):**
 - OS: Arco linux
 - terminal: alacritty
 - Version of Neovim: v0.5.1

**Additional context**
If you make this change in others.lua:

```
require("luasnip/loaders/from_vscode").load { paths = chadrc_config.plugins.options.luasnip.snippet_path }
```

Then you must do something like this in chadrc for friendly-snippets to show up:

```
luasnip = {
         snippet_path = {"~/.local/share/nvim/site/pack/packer/opt/friendly-snippets", "./lua/custom/snippets"},
 },
```

I found that this fix in others.lua worked also:

```
require("luasnip/loaders/from_vscode").load { paths = chadrc_config.plugins.options.luasnip.snippet_path }
require("luasnip/loaders/from_vscode").load()
```

And then you can just do something like this in chadrc:

```
luasnip = {
         snippet_path = {"./lua/custom/snippets"},
 },
```

Neither solution seems particularly elegant to me, but I'm a lua noob so maybe you can figure it out

**Comment by siduck76** [comment]
@jrodal98 thanks for pointing this out! 

Does the following work for you?

```
 require("luasnip/loaders/from_vscode").load { paths = chadrc_config.plugins.options.luasnip.snippet_path }
```

**Comment by jrodal98** [comment]
> @jrodal98 thanks for pointing this out!
> 
> Does the following work for you?
> 
> ```
>  require("luasnip/loaders/from_vscode").load { paths = chadrc_config.plugins.options.luasnip.snippet_path }
> ```

Yeah, that works, I needed to add the path to friendly snippets in my chadrc though, otherwise that doesn't get loaded anymore

```
luasnip = {
         snippet_path = {"~/.local/share/nvim/site/pack/packer/opt/friendly-snippets", "./lua/custom/snippets"},
 },

```

**Comment by siduck76** [comment]
> > @jrodal98 thanks for pointing this out!
> > Does the following work for you?
> > ```
> >  require("luasnip/loaders/from_vscode").load { paths = chadrc_config.plugins.options.luasnip.snippet_path }
> > ```
> 
> Yeah, that works, I needed to add the path to friendly snippets in my chadrc though, otherwise that doesn't get loaded anymore
> 
> ```
> luasnip = {
>          snippet_path = {"~/.local/share/nvim/site/pack/packer/opt/friendly-snippets", "./lua/custom/snippets"},
>  },
> ```

yes by default the path is empty , btw why the full path of friendly-snippets too?

**Comment by macarthy** [comment]
You might want to change the 

```  require("luasnip/loaders/from_vscode").load  ``` 

to 

```  

-- You can also use lazy loading so you only get in memory snippets of languages you use
require("luasnip/loaders/from_vscode").lazy_load( ... )  

```

https://github.com/L3MON4D3/LuaSnip/blob/800e1876df24a178252520253eb63cb5c9f0e1b9/Examples/snippets.lua#L494

**Comment by siduck76** [comment]
> You might want to change the
> 
> ` require("luasnip/loaders/from_vscode").load `
> 
> to
> 
> ```
> 
> -- You can also use lazy loading so you only get in memory snippets of languages you use
> require("luasnip/loaders/from_vscode").lazy_load( ... )  
> ```
> 
> https://github.com/L3MON4D3/LuaSnip/blob/800e1876df24a178252520253eb63cb5c9f0e1b9/Examples/snippets.lua#L494

we need the value of the variable from chadrc tho ._.

**Comment by macarthy** [comment]
you can pass the paths as normal 

``` require("luasnip/loaders/from_vscode").lazy_load() -- You can pass { paths = "./my-snippets/"} as well ```

**Comment by siduck76** [comment]
@jrodal98 @macarthy is this okay?
```
 -- others.lua
 
 luasnip = {
         snippets = { paths = "./my-snippets/" },
  }
  ````
  
  ```
     -- chadrc.lua
     require("luasnip/loaders/from_vscode").load { chadrc_config.plugins.options.luasnip.snippets}
  ```

**Comment by jrodal98** [comment]
Whoops, forgot about this and lost it in my feed

I actually don't think this will work. If you do this, then only the snippets at ./my-snippets/ will be loaded, NOT the friendly-snippets. In order to get both, I had to do one of the two solutions mentioned in my original post in the "additional context" section. That's why I included the full path to friendly snippets.

**Comment by neevek** [comment]
I have tried this:
```lua
M.plugins = {
  options = {   
    luasnip = {
      snippet_path = { "~/.config/nvim/lua/custom/vs_snippets" },
    },
  },
}
```
and this:
```lua
M.plugins = {
  options = {   
    luasnip = {
      snippet_path = "~/.config/nvim/lua/custom/vs_snippets",
    },
  },
}
```

I have already updated to the latest NvChad, but none of these settings works for me, my custom snippets are not loaded, what should I do to make it work?

**Comment by siduck** [comment]
@neevek did u try full path? like /home/neevek/.config ...?

**Comment by neevek** [comment]
@siduck I am on Mac, I tried `/Users/neevek/.config ...`, didn't work.

**Comment by siduck** [comment]
Idk then :/ . I dont use snippets so cant help. Can you try it with a base nvim config?

**Comment by neevek** [comment]
It seems `luasnip` only works with custom `snipmate` snippets in this case, but not `vscode` snippets, I tried adding the following line in `custom/init.lua` to load custom `snipmate` snippets, it works:
```lua
require("luasnip.loaders.from_snipmate").lazy_load({ paths = "~/.config/nvim/lua/custom/snippets" })
```
If I change `from_snipmate` to `from_vscode` and try loading `vscode` snippets, it doesn't work:
```lua
require("luasnip.loaders.from_vscode").lazy_load({ paths = "~/.config/nvim/lua/custom/vs_snippets" })
```

**Comment by jrodal98** [comment]
VScode style snippets work for me. Disclaimer I haven't updated in like two months, maybe more.


This is the relevant portion of my directory structure:

```
lua
├── custom
│   ├── chadrc.lua
│   └── snippets
│       ├── package.json
│       └── python.json
```
I have this in my chadrc:

```
luasnip = {
         snippet_path = { "./lua/custom/snippets" },
},
```

This is my package.json:

``` 
{
  "name": "JSR snippets",
  "engines": {
    "vscode": "^1.11.0"
  },
  "contributes": {
    "snippets": [
      {
        "language": "python",
        "path": "./python.json"
      }
    ]
  }
}
```

This is my python.json

```
{
    "shebang": {
        "prefix": "shebang",
        "body": ["#!/usr/bin/env python3", "# www.jrodal.com", "", "$0"],
        "description" : "Adds shebang line for default python 3 interpreter."
    }
}
```

**Comment by neevek** [comment]
@jrodal98 Maybe recent updates of `luasnip` broke it, I can confirm that it's an issue of `luasnip`, I am going to raise an issue in its repo.

**Comment by siduck** [comment]
> @jrodal98 Maybe recent updates of `luasnip` broke it, I can confirm that it's an issue of `luasnip`, I am going to raise an issue in its repo.
> 
> With the following commands, I can see my snippets are loaded.
> 
> ```lua
> :lua require("luasnip.loaders.from_snipmate").load({ paths = "~/.config/nvim/lua/custom/snippets" })
> :lua print(vim.inspect(require('luasnip.loaders._caches')))
> ```
> 
> With the following, the caches are empty:
> 
> ```lua
> :lua require("luasnip.loaders.from_vscode").load({ paths = "~/.config/nvim/lua/custom/vs_snippets" })
> :lua print(vim.inspect(require('luasnip.loaders._caches')))
> ```

so loaders.from_vscode stuff is outdated?

**Comment by neevek** [comment]
> > @jrodal98 Maybe recent updates of `luasnip` broke it, I can confirm that it's an issue of `luasnip`, I am going to raise an issue in its repo.
> > With the following commands, I can see my snippets are loaded.
> > ```lua
> > :lua require("luasnip.loaders.from_snipmate").load({ paths = "~/.config/nvim/lua/custom/snippets" })
> > :lua print(vim.inspect(require('luasnip.loaders._caches')))
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > With the following, the caches are empty:
> > ```lua
> > :lua require("luasnip.loaders.from_vscode").load({ paths = "~/.config/nvim/lua/custom/vs_snippets" })
> > :lua print(vim.inspect(require('luasnip.loaders._caches')))
> > ```
> 
> so loaders.from_vscode stuff is outdated?

I checked `~/.local/share/nvim/site/pack/packer/opt/LuaSnip`, which was already updated to the latest.

**Comment by neevek** [comment]
> @jrodal98 Maybe recent updates of `luasnip` broke it, I can confirm that it's an issue of `luasnip`, I am going to raise an issue in its repo.
> 
> With the following commands, I can see my snippets are loaded.
> 
> ```lua
> :lua require("luasnip.loaders.from_snipmate").load({ paths = "~/.config/nvim/lua/custom/snippets" })
> :lua print(vim.inspect(require('luasnip.loaders._caches')))
> ```
> 
> With the following, the caches are empty:
> 
> ```lua
> :lua require("luasnip.loaders.from_vscode").load({ paths = "~/.config/nvim/lua/custom/vs_snippets" })
> :lua print(vim.inspect(require('luasnip.loaders._caches')))
> ```

My bad, the snippets are not seen in the caches, but are actually loaded, I can use them after calling `load()`.
The root cause of the problem is that I didn't get it right on the vscode snippets format (JSON format), I added a comma at the last field of a JSON object, remvoing that comma fixed the problem.
